### PR TITLE
Add MAM4-JAX as an in-process box model

### PR DESCRIPTION
**Pull request by aliakherati**
## Summary

Adds [MAM4-JAX](https://github.com/reflective-org/MAM4-JAX) as a supported box model: a JAX implementation of the 4-mode MAM4 — the same fixed-structure modal scheme as the executable MAM4 AMBRS already drives, but as a differentiable Python library with no binaries to build.

Like TOMAS-JAX (#99) it runs **in process**: `create_input` → `run`/`run_ensemble` → the standard `analysis.Output`, plus a module-level `retrieve_model_state` mirroring the PartMC/MAM4 functions. Three commits; full details in [`MAM4_JAX.md`](MAM4_JAX.md), runnable walkthrough with plots in [`demo_mam4_jax.ipynb`](demo_mam4_jax.ipynb) (works straight after `pip install -r requirements.txt`; outputs stripped).

## Design

**No changes to shared framework code** — one new module plus one line in `ambrs/__init__.py` and one in `requirements.txt`.

- `create_input` builds MAM4's tracer array `q` from a 4-mode `Scenario`: number as #/kg-dry-air at `NUMPTR_AMODE`, per-species mass as kg/kg at `LMASSPTR_AMODE`, with the mode volume built from MAM4's *fixed* per-mode width so `calcsize` preserves number and diameter as a fixed point on step 0. Species/gas conventions follow `ambrs.mam4` (gases are kg/kg mass mixing ratios).
- `run` steps a cached JIT-compiled `calcsize → wateruptake → amicphys` with the `mdo_*` toggles implied by `AerosolProcesses`; `run_ensemble` reuses one compiled step across a whole ensemble.
- The final state becomes a `part2pop` binned-lognormal population (per-mode number, diameter, fixed GSD, per-species dry-mass fractions) plus a `GasMixture` (H2SO4, SO2, SOAG), so `compute_variable`/`nmae`/`kl_divergence` work unchanged.

## Physics notes reviewers may care about

- **Humidity enters through the water-vapour tracer.** MAM4-JAX's `wateruptake` derives RH from `q[0]`, not from the state's `relhum` field (only nucleation reads that). The adapter sets `q[0] = RH·qsat(T,p)` — reproducing the Fortran driver's reference IC — without which the aerosol runs dry at any scenario RH.
- **Fixed mode widths:** a scenario GSD that differs from MAM4's fixed per-mode sigma warns and is discarded.
- **Composition is lumped where MAM4 has no slot** (this MAM4-MOM build has no ammonium or nitrate anywhere; the Aitken mode has no BC/dust slot; primary carbon has no sulfate slot) — mass-conserving, with a warning naming the species. Full table in the docs.
- **Clear-sky only** (MAM4-JAX's cloudy path is unimplemented; `cldn` fixed at 0). `zmid`/`pblh` for PBL nucleation are constructor settings.
- `gas_phase_chemistry` maps to MAM4-JAX's built-in H2SO4 production term (injected inside gas-aerosol exchange, so it needs condensation on; warned otherwise). Because MAM4-JAX reads that configuration at JIT trace time, the rate is part of the compiled-step cache key.

## Testing

26 new tests in `test/test_mam4_jax.py`, same `unittest` style as the suite:

- **With mam4-jax installed:** `102 passed` (76 baseline + 26 new).
- **Without it:** `76 passed, 26 skipped` — `import ambrs` still works; the guard also catches `OSError` so a broken partial install can't take it down.

Physics invariants covered: exact number round-trip through air density, mass/volume consistency with the fixed-width lognormal, `q[0]` matching `rh·qsat`, all-processes-off preserving total number, coagulation reducing it.

## Dependency note

mam4-jax is public and installs over `git+https` (its wheel ships the `_coag_tables.npz` data file as of reflective-org/MAM4-JAX#62), so it's declared in `requirements.txt` and CI will exercise the tests. It pulls in `jax`/`jaxlib`/`diffrax`. If you'd rather keep those optional, say the word and I'll move it to an optional requirements file — the import guard already makes the tests skip cleanly.

**Overlap note:** this PR and #99 each add one line to `ambrs/__init__.py`, `requirements.txt` and the README's model list; whichever merges second needs a trivial rebase.

Suggested labels: `box model integration`, `enhancement`, `python`, `dependencies`, `testing`, `mam4`.

### CODE REVIEW

**Comment by GlazerMann** [comment]
You might want to freeze these.   Just a thought.

diff --git a/requirements.txt b/requirements.txt
--- a/requirements.txt
+++ b/requirements.txt
@@ -24 +24 @@
-mam4-jax @ git+https://github.com/reflective-org/MAM4-JAX.git@main
+mam4-jax @ git+https://github.com/reflective-org/MAM4-JAX.git@5ed465661dcd1b4a858ed92524ac58fc37696cfd
diff --git a/ambrs/mam4_jax.py b/ambrs/mam4_jax.py
--- a/ambrs/mam4_jax.py
+++ b/ambrs/mam4_jax.py
@@ -305 +305 @@
-                '    pip install "mam4-jax @ git+https://github.com/reflective-org/MAM4-JAX.git@main"')
+                '    pip install "mam4-jax @ git+https://github.com/reflective-org/MAM4-JAX.git@5ed465661dcd1b4a858ed92524ac58fc37696cfd"')
diff --git a/MAM4_JAX.md b/MAM4_JAX.md
--- a/MAM4_JAX.md
+++ b/MAM4_JAX.md
@@ -24 +24 @@
-pip install "mam4-jax @ git+https://github.com/reflective-org/MAM4-JAX.git@main"
+pip install "mam4-jax @ git+https://github.com/reflective-org/MAM4-JAX.git@5ed465661dcd1b4a858ed92524ac58fc37696cfd"

**Comment by GlazerMann** [comment]
I think there is a gas-unit bug in the conversion from the MAM4-JAX result back into an AMBRS `GasMixture`.

Upstream MAM4-JAX documents the gas entries in `q` as mol/mol-air; H2SO4 is explicitly reported that way in its reference schema. That makes assigning AMBRS mole ratios directly into the gas entries of `q` reasonable.

However, after the run this adapter does:

```python
gas_mixture = build_gas_mixture({
    'H2SO4': float(q[_H2SO4_IDX]),
    'SO2': float(input.so2_mmr),
    'SOAG': float(q[_SOAG_IDX]),
    'units': 'kg_per_kg',
})
```

`build_gas_mixture()` interprets `kg_per_kg` by multiplying each value by `MW_air / MW_gas` to obtain mole ratio. If the MAM4-JAX gas values are already mol/mol, that conversion changes them incorrectly.

Should this be:

```python
'units': 'mole_ratio'
```

instead?

The PR description also currently says MAM4-JAX gases are kg/kg, which appears inconsistent with upstream MAM4-JAX's documented gas representation.

A strong regression would initialize known SO2/H2SO4/SOAG mole ratios, run with all processes disabled, and verify that the resulting `GasMixture.mole_ratio` preserves those values.
